neco-exporter/pkg/collector/cluster/ciliumid/identity_watcher.go

```diff
@@ -0,0 +1,138 @@
+package ciliumid
+
+import (
+	"context"
+	"fmt"
+	"log/slog"
+	"maps"
+	"sync"
+	"time"
+
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/tools/cache"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	indexKey = "neco-exporter.ciliumid.namespace"
+)
+
+func newCiliumIdentity() *unstructured.Unstructured {
+	u := &unstructured.Unstructured{}
+	u.SetGroupVersionKind(schema.GroupVersionKind{
```

Comment:
Are there any reasons why you don't use directly `CiliumIdentity` type from `github.com/cilium/cilium/pkg/k8s/apis/cilium.io/v2`?

FYI
We use this package in `cep-checker`.
https://github.com/cybozu/neco-containers/blob/a455a87bca5a83ff039b4f13434d51bdd4ec893b/cep-checker/main.go#L196